helpers/migrate.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import copy
+import subprocess
+import sys
+import shutil
+import re
+
+
+class GSuiteMigration:
+    """
+    Migrate the resources from a flat project factory to match the new
+    module structure created by the G Suite refactor.
+    """
+
+    CORE_PROJECT_FACTORY_SELECTORS = [
+        ("google_organization", "org"),
+        ("google_compute_default_service_account", "default"),
+        ("null_data_source", "data_final_group_email"),
+        ("null_data_source", "data_group_email_format"),
+        ("google_compute_shared_vpc_service_project", None),
+        ("google_compute_subnetwork_iam_member", None),
+        ("google_project", None),
+        ("google_project_iam_member", None),
```

Comment:
`google_project_iam_member.gsuite_group_role` and `google_project_iam_member.controlling_group_vpc_membership` remain in `gsuite_enabled` so this rule must be restricted.

In case we test this script against test state, `google_project_iam_member.additive_sa_role` and `google_project_iam_member.additive_shared_vpc_role` remain in `full` fixture.

Reply:
The refactoring of the shared VPC `roles/compute.networkUser` role means that we can't perform a simple migration through `terraform state mv`. I'm getting this diff:

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create
-/+ destroy and then create replacement

Terraform will perform the following actions:

  + module.project-factory.google_project_iam_member.controlling_group_vpc_membership
      id:      <computed>
      etag:    <computed>
      member:  "group:pf-gsuite-migrate-group@phoogle.net"
      project: "thebo-host-c15e"
      role:    "roles/compute.networkUser"

-/+ module.project-factory.module.project-factory.google_project_iam_member.controlling_group_vpc_membership[1] (new resource required)
      id:      "thebo-host-c15e/roles/compute.networkUser/group:pf-gsuite-migrate-group@phoogle.net" => <computed> (forces new resource)
      etag:    "BwV872YZKPU=" => <computed>
      member:  "group:pf-gsuite-migrate-group@phoogle.net" => "serviceAccount:303471683246@cloudservices.gserviceaccount.com" (forces new resource)
      project: "thebo-host-c15e" => "thebo-host-c15e"
      role:    "roles/compute.networkUser" => "roles/compute.networkUser"

-/+ module.project-factory.module.project-factory.google_project_iam_member.controlling_group_vpc_membership[2] (new resource required)
      id:      "thebo-host-c15e/roles/compute.networkUser/serviceAccount:303471683246@cloudservices.gserviceaccount.com" => <computed> (forces new resource)
      etag:    "BwV872YZKPU=" => <computed>
      member:  "serviceAccount:303471683246@cloudservices.gserviceaccount.com" => "serviceAccount:project-service-account@pf-gsuite-migrate-group-23dc.iam.gserviceaccount.com" (forces new resource)
      project: "thebo-host-c15e" => "thebo-host-c15e"
      role:    "roles/compute.networkUser" => "roles/compute.networkUser"
```

I think we're stuck with re-creating these resources or doing some serious state surgery, which increases the risk of something going wrong.

Reply:
Is there a risk in recreating those resources?